src/main/java/org/apache/commons/imaging/formats/tiff/TiffRasterData.java

```diff
@@ -37,37 +35,45 @@
     public TiffRasterData(final int width, final int height) {
         if (width <= 0 || height <= 0) {
             throw new IllegalArgumentException(
-                "Raster dimensions less than or equal to zero are not supported");
+                    "Raster dimensions less than or equal to zero are not supported");
         }
-        final int nCells = width * height;
-        data = new float[nCells];
         this.width = width;
         this.height = height;
+    }
 
+    protected final int checkCoordinates(final int x, final int y) {
+        if (x < 0 || x >= width || y < 0 || y >= height) {
+            throw new IllegalArgumentException(
+                    "Coordinates out of range (" + x + ", " + y + ")");
+        }
+        return y * width + x;
```

Comment:
Could we add a Javadoc explaining what this method does? I thought it would just check the coordinates, verifying whether they are valid or not. But looks like it's actually doing that and also returning the `y` coordinate times the `width`... plus `x`? Even though we check `height` in the if-statement above we never use it in `checkCoordinates`, is that correct?

Reply:
basically, done.  changed the name of the method to the more descriptive checkCoordinatesAndComputeIndex().